File: leetcode/problem_102.py
### Problem 102. 
from collections import deque

class Solution:
    def levelOrder(self, root: Optional[TreeNode]) -> List[List[int]]:
        q = deque()
        result = []
        q.append(root)
        if(root is None):
            return result
        while(True):
            size = len(q)
            if(size == 0):
                return result
            data = []
            while(size > 0):
                temp = q.popleft()
                data.append(temp.val)
                if(temp.left):
                    q.append(temp.left)
                if(temp.right):
                    q.append(temp.right)
                size -=1 
            result.append(data)
        return result

File: leetcode/problem_102.kt
// Problem 102. Binary Tree Level Order Traversal
// tags: Medium, Tree, BFS

class Solution {
    fun levelOrder(root: TreeNode?): List<List<Int>> {
        val result: MutableList<MutableList<Int>> = mutableListOf()
        fun levelTrav(node: TreeNode?, level:Int){
            if(node == null) return
            if(result.size == level){
                result.add(mutableListOf<Int>())
            }
            result[level].add(node!!.`val`)
            if(node.`left` != null){
                levelTrav(node.`left`, level + 1)
            }
            if(node.`right`!= null){
                levelTrav(node.`right`, level + 1)
            }
        }

        levelTrav(root, 0)
        return result
    }
}